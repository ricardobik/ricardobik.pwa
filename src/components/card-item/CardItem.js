/* eslint-disable jsx-a11y/no-static-element-interactions */

import React from 'react';
import cx from 'classnames';

import logo from '../../images/svgs/logo.svg';
import back from '../../images/svgs/back.svg';
import styles from './styles.scss';

const onClick = (value, openedCard, toggleCard) => {
  if (openedCard) {
    return toggleCard(null);
  }
  return toggleCard(value);
};

const getContainerStyles = (openedCard) => {
  if (openedCard) {
    return cx(styles.container, styles.opened);
  }
  return styles.container;
};

const renderBrandLogo = (openedCard) => {
  if (openedCard) {
    return (
      <figure className={styles.brandLogo}>
        <img src={logo} alt="Brand logo" />
      </figure>
    );
  }
  return null;
};

const renderBackButton = (openedCard) => {
  if (openedCard) {
    return (
      <figure className={styles.backButton}>
        <img src={back} alt="Back button" />
      </figure>
    );
  }
  return null;
};

const CardItem = ({ value, openedCard, toggleCard }) => (
  <div
    onClick={() => onClick(value, openedCard, toggleCard)}
    className={getContainerStyles(openedCard)}
  >
    <span>{value}</span>
    {renderBrandLogo(openedCard)}
    {renderBackButton(openedCard)}
  </div>
);

CardItem.defaultProps = {
  openedCard: null,
};

CardItem.propTypes = {
  value: React.PropTypes.string.isRequired,
  openedCard: React.PropTypes.string,
  toggleCard: React.PropTypes.func.isRequired,
};

export default CardItem;
