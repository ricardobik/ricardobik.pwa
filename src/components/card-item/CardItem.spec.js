import React from 'react';
import renderer from 'react-test-renderer';

import CardItem from './CardItem';

const props = {
  value: 'value',
  toggleCard: () => {},
};

test('Snapshot', () => {
  const component = renderer.create(
    <CardItem {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
