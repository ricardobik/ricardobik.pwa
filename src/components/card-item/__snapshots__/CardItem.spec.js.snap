// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <span>
    value
  </span>
</div>
`;
